### ISSUE TITLE
error: failed to get `axum` as a dependency of package `svix-server v0.74.0 (/app/svix/server/v0.74/svix-server)`

### ISSUE THREAD

**Issue by dipk-mish7** [svix/svix-webhooks#914]
We have setup Cargo repository in artifactory as github is blocked in our org and while doing cargo build it fails to fetch dependency from github.
If I clone this project in my local machine in ubuntu and I do cargo build at the path svix/server/v0.74 it works fine for me and is able to build the package. I get target folder created and Cargo.lock file created as well. Please find below.
![image](https://user-images.githubusercontent.com/52923075/236994529-d3f6c562-c769-46b3-818c-a3116bf8d07f.png)
![image](https://user-images.githubusercontent.com/52923075/236995822-e94c1a63-c3a7-4fda-9f31-e54068f8bc96.png)

Now I need to setup a gitlab pipline and build this in there so I am trying to build the  Dockerfile located here https://github.com/svix/svix-webhooks/blob/main/server/Dockerfile I have whitelisted the image **rust:1.65-slim-bullseye** so that it can be accessible from organization network. When I spin a docker container using **rust:1.65-slim-bullseye** and bind my project inside that and do cargo build it throws me below error. I installed curl in the docker container to check if the package is facing any connectivity issue. To my surprise curl was able to fetch the dependency.  Not sure why I am not able to do cargo build in this container with image rust:1.65-slim-bullseye. I will be building this dockerfile to craete an image which can be further deployed in our environments. I also checked the cargo.lock file which gets generated after build doesn't have auxm as package listed there in my local machine.
![image](https://user-images.githubusercontent.com/52923075/236996051-bafa646c-1ace-48d6-8459-e280f6880370.png)

I have placed credentials.toml and config.toml(with registory pointing to artifactory) in the .cargo directory so that it can point towards the artifactory. For some reasons it points to artifactory nicely but fails to download auxm.

**Comment by dipk-mish7** [comment]
Hey Tom.. thanks. I tried with rust:1.69.0 image and it worked fine. I was able to build the image locally. It was able to fetch auxm as dependency. Now the problem is in gitlab-ci this is not working. It timesout. This error came in local and I did cargo login --registry artifactory. After pasting token it generated credential.toml in the CARGO_HOME directory. I copied credentials.toml  in my dockerfile in /usr/localcargo/ to ensure when dockerimage is build it has that credentials in place. This is working in local but not in gitlab-ci
![image](https://github.com/svix/svix-webhooks/assets/52923075/e8c45cd3-be76-4ff5-a6d9-0b6b88b42962)
